modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/components/SimpleInputModal.es.js

```diff
@@ -161,9 +175,12 @@ const SimpleInputModal = ({
 								disabled={loadingResponse}
 								id={`${namespace}${mainFieldName}`}
 								name={`${namespace}${mainFieldName}`}
-								onChange={(event) =>
-									setInputValue(event.target.value)
-								}
+								onChange={(event) => {
+									if (errorMessage) {
+										setErrorMessage('');
+									}
```

Comment:
There may be a bug here. What happens when you delete the last letter in input? I'd expect that we show an error, but with this code `onChange` never sets an error.

Reply:
Why should it show an error when deleting the last letter of the input? you are deciding the value of the input, as long as the value is valid there should be no problem, right? 🤔

Reply:
If the field is required, it should should an error when input becomes empty, shouldn't it?

Reply:
You are setting a precedent here that change of input changes validation status, by removing an error. It only makes sense to me that change of input can also change validation status to error.

Reply:
If the user is already making changes shouldn't we delete the existing error because until the user resubmits the form we can't tell if it is happening again?

Reply:
If this is not a bug, but intended behavior, this is a product / UX call. @ugeortiz you may want to review this, as well as the odd error handling on request failure. In that's the case, this is not a blocker for PR. It _is_ an improvement to current behavior, that we may want to further improve in a followup.

Reply:
If it is not a blocker for this pr, how about if I fix the "_" of the handleSummit to pass this pr and we open a research task to see if it is worth to unify it? 😄

Reply:
> If the user is already making changes shouldn't we delete the existing error because until the user resubmits the form we can't tell if it is happening again?

In my opinion, yes, this is good behavior. For sure we want individual inline form validation, [that is in Lexicon](https://liferay.design/lexicon/core-components/forms/forms-validation/). What I'm saying is this PR is missing validation behavior _in addition_ to that one, where we set validation error when required field becomes empty.

Reply:
Oh, okay, for that change, do you want me to return that the field is empty or also put that it is required (thanks for the link!)

Reply:
I'm not sure what you mean. See how we did validation [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L619-L622). In the linked file you'll see that we're validating on blur, instead of on change like in this PR, but on change may be better. I'm not sure, that's a detail that's not really strictly defined :sweat_smile:

Reply:
@kresimir-coko Please folluwup on this (I'll be on PTO next two weeks).